client/java/src/main/java/io/openlineage/client/dataset/hostResolver/DatasetHostResolverBuilder.java

```diff
@@ -0,0 +1,15 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.dataset.hostResolver;
+
+public interface DatasetHostResolverBuilder {
```

Comment:
Maybe a better name would be "Dataset**Namespace**Resolver" to keep up with the already established vocabulary used within OpenLineage?